server/api/routers/inference/inference.py

```diff
@@ -28,19 +28,19 @@ async def chat(
         if not session_id:
             session_id = str(uuid.uuid4())
 
-        # Use ChatProcessor directly with the full OpenAI-style request
-        response_message, _tool_info = ChatProcessor.process_chat(
-            request, session_id
-        )
-
-        # Set session header for client continuity
-        set_session_header(response, session_id)
-
-        # If client requested streaming, return Server-Sent Events stream
+        # If client requested streaming, return Server-Sent Events stream using agent-native streaming when possible
         if request.stream:
-            return create_streaming_response(request, response_message, session_id)
-
-        # Return OpenAI-compatible response
+            stream_iter = ChatProcessor.stream_chat(request, session_id)
```

Comment:
I would love to name these two ChatProcessor methods so that they are clearly related like `ChatProcessor.process_chat_streaming` and `ChatProcessor.process_chat`